script-test/playbackstrategies/modifiers/html5tests.js

```diff
@@ -0,0 +1,3260 @@
+    function (Squire, MediaPlayerBase) {
+      describe('HTML5 Base', function () {
+        var injector = new Squire();
+        var html5Player;
+        var sourceContainer;
+        var player;
+        var mockSourceElement;
+        var mockVideoMediaElement;
+        var mockAudioMediaElement;
+        var metaDataCallback;
+        var finishedBufferingCallback;
+        var errorCallback;
+        var endedCallback;
+        var timeupdateCallback;
+        var waitingCallback;
+
+        var recentEvents = [];
+
+        var logger = jasmine.createSpyObj('logger', ['warn', 'debug', 'error']);
+
+        mockSourceElement = document.createElement('source');
+        mockVideoMediaElement = document.createElement('video');
+        mockAudioMediaElement = document.createElement('audio');
+
+        function playFromNearCurrentTime (currentTime, targetTime) {
+          player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'http://url/', 'video/mp4', sourceContainer, {});
+
+          player.beginPlaybackFrom(0);
+          metaDataCallback();
+          finishedBufferingCallback();
+
+          expect(player.getState()).toEqual(MediaPlayerBase.STATE.PLAYING);
+
+          waitingCallback();
+
+          mockVideoMediaElement.currentTime = currentTime;
+
+          expect(player.getState()).toEqual(MediaPlayerBase.STATE.BUFFERING);
+
+          player.playFrom(targetTime);
+          finishedBufferingCallback();
+
+          expect(player.getState()).toEqual(MediaPlayerBase.STATE.PLAYING);
+        }
+
+        function eventCallbackReporter (event) {
+          recentEvents.push(event.type);
+        }
+
+        function giveMediaElementMetaData (mediaElement, metadata) {
+          try {
+            spyOnProperty(mediaElement, 'seekable').and.returnValue(
+              {
+                start: function () {
+                  return metadata.start;
+                },
+                end: function () {
+                  return metadata.end;
+                },
+                length: 2
+              });
+          } catch (ex) {
+            mediaElement.seekable = {
+              start: function () {
+                return metadata.start;
+              },
+              end: function () {
+                return metadata.end;
+              },
+              length: 2
+            };
+          }
+        }
+
+        beforeEach(function (done) {
+          mockSourceElement = document.createElement('source');
+          mockVideoMediaElement = document.createElement('video');
+          mockAudioMediaElement = document.createElement('audio');
+
+          logger = jasmine.createSpyObj('logger', ['warn', 'debug', 'error']);
+          sourceContainer = document.createElement('div');
+
+          injector.require(['bigscreenplayer/playbackstrategy/modifiers/html5'], function (SquiredHtml5) {
+            html5Player = SquiredHtml5;
+            done();
+          });
+        });
+
+        describe('tests', function () {
```

Comment:
Unnecessary `describe` block. If removed, you could also combine the `beforeEach`s.